backend/lucene/src/main/java/org/hibernate/search/backend/lucene/logging/impl/Log.java

```diff
@@ -380,21 +380,22 @@ SearchException unableToCloseIndexWriterAfterFailures(String causeMessage, @Para
 	SearchException invalidDecimalScale(Integer decimalScale, @Param EventContext eventContext);
 
 	@Message(id = ID_OFFSET + 84,
-			value = "Invalid search predicate: '%1$s'. You must build the predicate from a scope targeting indexes %3$s,"
-					+ " but the given predicate was built from a scope targeting indexes %2$s.")
+			value = "Invalid search predicate: '%1$s'. You must build the predicate from a scope targeting indexes %3$s or a superset of them,"
+					+ " but the given predicate was built from a scope targeting indexes %2$s, where %4$s indexes are missing from the predicate scope.")
 	SearchException predicateDefinedOnDifferentIndexes(SearchPredicate predicate, Set<String> predicateIndexes,
-			Set<String> scopeIndexes);
+			Set<String> scopeIndexes, Set<String> scopeDifference);
 
 	@Message(id = ID_OFFSET + 85,
-			value = "Invalid search sort: '%1$s'. You must build the sort from a scope targeting indexes %3$s,"
-					+ " but the given sort was built from a scope targeting indexes %2$s.")
-	SearchException sortDefinedOnDifferentIndexes(SearchSort predicate, Set<String> predicateIndexes, Set<String> scopeIndexes);
+			value = "Invalid search sort: '%1$s'. You must build the sort from a scope targeting indexes %3$s or a superset of them,"
+					+ " but the given sort was built from a scope targeting indexes %2$s, where %4$s indexes are missing from the sort scope.")
+	SearchException sortDefinedOnDifferentIndexes(SearchSort predicate, Set<String> predicateIndexes, Set<String> scopeIndexes,
+			Set<String> scopeDifference);
 
 	@Message(id = ID_OFFSET + 86,
-			value = "Invalid search projection: '%1$s'. You must build the projection from a scope targeting indexes %3$s,"
-					+ " but the given projection was built from a scope targeting indexes %2$s.")
+			value = "Invalid search projection: '%1$s'. You must build the projection from a scope targeting indexes %3$s or a superset of them,"
```

Comment:
It's weird that you didn't have to do the same for aggregations, no?

And highlighters, maybe -- though those may be different, I guess?